vol_object_test.c

```diff
@@ -5309,6 +5380,30 @@ test_object_visit(void)
         }
         PART_END(H5Ovisit_dtype);
 
+        PART_BEGIN(H5Ovisit_attr)
+        {
+            TESTING_2("H5Ovisit on an attribute")
+
+            i = 0;
+
+            if (H5Ovisit3(attr_id, H5_INDEX_CRT_ORDER, H5_ITER_INC, object_visit_simple_callback, &i,
+                          H5O_INFO_ALL) < 0) {
+                H5_FAILED();
+                HDprintf("    H5Ovisit on an attribute failed!\n");
+                PART_ERROR(H5Ovisit_attr);
+            }
+
+            /* Should have same effect as calling H5Ovisit on group_id */
+            if (i != OBJECT_VISIT_TEST_NUM_OBJS_VISITED) {
```

Comment:
If H5Ovisit is actually visiting the object the attribute is attached to, you may want to consider filing a GitHub issue to change the documentation for the function since it currently says that there will not be iteration for objects other than groups.

Reply:
I think this is consistent with the documentation - ["If obj_id is an attribute identifier, then the object where the attribute is attached will be iterated."](https://docs.hdfgroup.org/hdf5/develop/group___h5_o.html#title39)

Reply:
Hmm ok, my documentation must have been behind from a commit yet to be merged